### PR TITLE
Feature test

### PR DESCRIPTION
**Pull request by jitendra2329**
Unit test case for UserDB and UserRepo
and Integration test for UserRepo is added.

### CODE REVIEW

File: src/main/scala/App/Main.scala [hunk 1]
```diff
@@ -40,6 +41,11 @@ object Main extends App {
 
   println("-------- All Users Deleted ---------------")
   println(userRepo.getAll)
+
```
Comment (Bhavya-verma-knoldus) [inline src/main/scala/App/Main.scala:44]: Why have you added these extra lines?

Comment (jitendra2329) [reply src/main/scala/App/Main.scala:44]: These line were already in the code, but when I was reviewing my code I think then these lines may be touched or formated

File: src/test/scala/db/UserDBUnitTest.scala [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+import org.scalatest.flatspec.AnyFlatSpec
+import org.scalatestplus.mockito.MockitoSugar.mock
+import java.util.UUID
+import scala.collection.mutable.ListBuffer
+
+class UserDBUnitTest extends AnyFlatSpec {
+
+  val userDB: DAO = mock[DAO]
+
+  it should "Add user should return new user added." in {
```
Comment (Bhavya-verma-knoldus) [inline src/test/scala/db/UserDBUnitTest.scala:16]: `it` is always used from the second test case. First test case always describes completely what is the intent.

File: src/main/scala/App/Main.scala (hunk 1 above)
Comment (jitendra2329) [reply src/main/scala/App/Main.scala:44]: Here, I was just checking something for the UserDB mocking

File: src/test/scala/db/UserDBUnitTest.scala [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    val user = Users(UUID.randomUUID(), "Jeet", 23, "gkp", "12/2/1998", Customer)
+    when(userDB.addUser(user)).thenReturn("new user added.")
```
Comment (Bhavya-verma-knoldus) [inline src/test/scala/db/UserDBUnitTest.scala:18]: We do not use mocking if there are no dependent methods called.

File: src/test/scala/service/UserRepoIntegrationTest.scala [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+import scala.collection.mutable.ListBuffer
+
+class UserRepoIntegrationTest extends AnyFlatSpec {
+
+  val userDB = new UserDB
+  val userRepo = new UserRepo(userDB)
+  val customerUserId: UUID = UUID.randomUUID()
+  val adminUserId: UUID = UUID.randomUUID()
+
+  it should "return 'new user added.'" in {
```
Comment (Bhavya-verma-knoldus) [inline src/test/scala/service/UserRepoIntegrationTest.scala:18]: Ditto: https://github.com/jitendra2329/user-management-service/pull/3/files#r1176132727

File: src/test/scala/db/UserDBUnitTest.scala (hunk 3 above)
Comment (jitendra2329) [reply src/test/scala/db/UserDBUnitTest.scala:18]: ok, so we should make object of that class, Right?
val userDB = new UserDB

File: src/main/scala/App/Main.scala (hunk 1 above)
Comment (Bhavya-verma-knoldus) [reply src/main/scala/App/Main.scala:44]: Kindly remove the extra lines added.

File: src/test/scala/db/UserDBUnitTest.scala (hunk 3 above)
Comment (jitendra2329) [reply src/test/scala/db/UserDBUnitTest.scala:18]: val userDB = new UserDB

  it should "Add user should return new user added." in {
    val user = Users(UUID.randomUUID(), "Jeet", 23, "gkp", "12/2/1998", Customer)
    assert(userDB.addUser(user) == "new user added.")
  }
  
  Will it be ok, for the above

**Pull request merged**